README.md

```diff
@@ -62,6 +62,74 @@ All articles are recorded in a sqlite database.
 
 In some cases, you may wish to suppress articles from being posted, even though they would otherwise match. You can do so by writing a new function, `check`, and placing it in a file named `blocklist.py` in the configuration directory. `check` takes an Article (and so has access to its `outlet`, `title`, and `url`) and should return `true` for any article that should be skipped.
 
+## Development
+
+### Quick Start
+
+```bash
+poetry install
+poetry run trackthenews sample_project
+# Follow the setup script instructions
+cat sample_project/matchlist-sample.txt > sample_project/matchlist.txt
+cat sample_project/rssfeeds-sample.json > sample_project/rssfeeds.json
+poetry run trackthenews sample_project
+```
+
+### Detailed Instructions
+
+To develop `trackthenews`, clone the repository and install the package using [poetry][] and run the CLI tool:
+
+```bash
+# This will create a virtual environment and install trackthenews and its dependencies
+poetry install --with=dev
+# This will run the setup script
+poetry run trackthenews sample_project
+```
+
+On first run this will take you through the interactive setup process. Follow the instructions. You will need a Twitter account with application keys and/or a Mastodon account.
+
+When the setup process is complete, the configuration for your bot will live in the `sample_project` directory. Fill out the `matchlist.txt`, `matchlist_case_sensitive.txt`, and `rssfeeds.json` files or copy configuration from the provided samples:
+
+```bash
+cat sample_project/matchlist-sample.txt > sample_project/matchlist.txt
+cat sample_project/rssfeeds-sample.json > sample_project/rssfeeds.json
+```
+
+`matchlist_case_sensitive.txt` is optional and we don't provide a sample.
+
+Now run the bot:
+
+```bash
+poetry run trackthenews sample_project
```

Comment:
This is not extremely pertinent to this PR, but it's something I noticed: this command stopped printing output part of the way through and I had to control+c it. This was the traceback:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/chigby/fpf/trackthenews/trackthenews/core.py", line 627, in main
    articles = parse_feed(outlet, url, delicate, redirects)
  File "/home/chigby/fpf/trackthenews/trackthenews/core.py", line 282, in parse_feed
    feed = feedparser.parse(url)
  File "/home/chigby/.cache/pypoetry/virtualenvs/trackthenews-h6XFDcUG-py3.9/lib/python3.9/site-packages/feedparser/api.py", line 216, in parse
    data = _open_resource(url_file_stream_or_string, etag, modified, agent, referrer, handlers, request_headers, result)
  File "/home/chigby/.cache/pypoetry/virtualenvs/trackthenews-h6XFDcUG-py3.9/lib/python3.9/site-packages/feedparser/api.py", line 115, in _open_resource
    return http.get(url_file_stream_or_string, etag, modified, agent, referrer, handlers, request_headers, result)
  File "/home/chigby/.cache/pypoetry/virtualenvs/trackthenews-h6XFDcUG-py3.9/lib/python3.9/site-packages/feedparser/http.py", line 171, in get
    f = opener.open(request)
  File "/usr/lib64/python3.9/urllib/request.py", line 517, in open
    response = self._open(req, data)
  File "/usr/lib64/python3.9/urllib/request.py", line 534, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/lib64/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.9/urllib/request.py", line 1375, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "/usr/lib64/python3.9/urllib/request.py", line 1350, in do_open
    r = h.getresponse()
  File "/usr/lib64/python3.9/http/client.py", line 1377, in getresponse
    response.begin()
  File "/usr/lib64/python3.9/http/client.py", line 320, in begin
    version, status, reason = self._read_status()
  File "/usr/lib64/python3.9/http/client.py", line 281, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib64/python3.9/socket.py", line 704, in readinto
    return self._sock.recv_into(b)
KeyboardInterrupt
```

Essentially, feedparser is hanging trying to fetch the feed. Looking a bit through the feedparser code, it seems like it's not possible to add some kind of http timeout. My recommendation here would be to use `requests` to fetch the feed, and then pipe that data into feedparser for actual parsing.

Reply:
Could be a chance to use `httpx` instead, which seems like it might be the way of the future 👀 

I think maybe it would be good to save that for another PR though, since this isn't specific to the Poetry transition.